### PR TITLE
[#460] Add dev_permission flag to bazel

### PR DESCRIPTION
**Pull request by elBoberido**
## Notes for Reviewer

1. Permissions are used to manage access to shared memory objects on the file system used by `iceoryx2`
2. In a docker environment, the user permissions on the host are not aligned with that within the container environment
3. A `dev_permissions` feature flag has been added, allowing all processes access to all shared memory objects to allow usage in the docker environment
4. This PR adds the `dev_permissions` feature flag to the `bazel` build

## Pre-Review Checklist for the PR Author

1. [x] Add sensible notes for the reviewer
1. [x] PR title is short, expressive and meaningful
1. [x] Relevant issues are linked in the [References](#references) section
1. [x] Every source code file has a copyright header with `SPDX-License-Identifier: Apache-2.0 OR MIT`
1. [x] Branch follows the naming format (`iox2-123-introduce-posix-ipc-example`)
1. [x] Commits messages are according to this [guideline][commit-guidelines]
    - [x] Commit messages have the issue ID (`[#123] Add posix ipc example`)
    - [x] Commit author matches [Eclipse Contributor Agreement][eca] (and ECA is signed)
1. [x] Tests follow the [best practice for testing][testing]
1. [x] Changelog updated [in the unreleased section][changelog] including API breaking changes
1. [x] Assign PR to reviewer
1. [x] All checks have passed (except `task-list-completed`)

[commit-guidelines]: https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html
[eca]: http://www.eclipse.org/legal/ECA.php
[testing]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/doc/website/concepts/best-practice-for-testing.md
[changelog]: https://github.com/eclipse-iceoryx/iceoryx2/blob/main/doc/release-notes/iceoryx2-unreleased.md

## Checklist for the PR Reviewer

- [x] Commits are properly organized and messages are according to the guideline
- [x] Unit tests have been written for new behavior
- [x] Public API is documented
- [x] PR title describes the changes

## Post-review Checklist for the PR Author

1. [x] All open points are addressed and tracked via issues

## References

Relates to #460

### CODE REVIEW

**Review by orecham — DISMISSED** [review]

File: doc/bazel/README.md [hunk 1]
```diff
@@ -152,6 +158,44 @@ cc_binary(
 * **For C++**: Use `@iceoryx2//:iceoryx2-cxx-static` or
   `@iceoryx2//:iceoryx2-cxx-shared`
 
+### Feature Flags
+
+iceoryx2 provides several feature flags that can be configured using bazel
+build options.These flags allow you to enable or disable specific features
```
Comment (orecham) [inline doc/bazel/README.md:164]:
```suggestion
build options. These flags allow you to enable or disable specific features
```

Comment (orecham) [reply doc/bazel/README.md:164]: Also some other issues the linter would catch. Did you install an auto-formatter?

Comment (elBoberido) [reply doc/bazel/README.md]: Is it part of a git hook?

Comment (orecham) [reply doc/bazel/README.md]: @elBoberido No but am not so sure if it is a good idea for that. Ideally the auto-corrections should be verified quickly (either with a git diff or in-editor with an auto formatter). There is to to the fixing from the terminal: https://github.com/eclipse-iceoryx/iceoryx2/blob/main/internal/scripts/check_markdown_linting.sh

**Review by orecham — APPROVED** [review]

File: doc/bazel/README.md (hunk 1 above)
Comment (elBoberido) [reply doc/bazel/README.md]: @orecham I know that script but forgot to run it. I'll create a git hook to run it on all markdown files in the commit if the markdown linter is available. That should be fairly fast and catches the issues immediately for regular contributors.

Comment (orecham) [reply doc/bazel/README.md]: @elBoberido If you want to do that, the script should already take care of finding the files on the current branch (including those not committed), so the hook would only need to run the script.

The hesitation I have is the time it takes (could be too long for a hook) and the line length correction which can mess up some intentional formatting.

Feel free to change the script however you want though.

Comment (elBoberido) [reply doc/bazel/README.md]: The hook would only do the check, not perform any changes. That would be up to the developer to trigger.

**Pull request merged**